clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -621,7 +628,6 @@ public void testPollTimerExpiration() {
         assertHeartbeat(heartbeatRequestManager, DEFAULT_HEARTBEAT_INTERVAL_MS);
         verify(membershipManager).transitionToSendingLeaveGroup(true);
         verify(heartbeatState).reset();
-        verify(heartbeatRequestState).reset();
         verify(membershipManager).onHeartbeatRequestSent();
 
         when(membershipManager.state()).thenReturn(MemberState.STALE);
```

Comment:
maybe no needed anymore? (the following expectation on `shouldSkipHeartbeat` is probably all we need)